### PR TITLE
Split debug-layer into smaller files

### PR DESCRIPTION
**Pull request by lucy96chen**
Changes:
- In the same vein as D3D/Vulkan/CUDA/cpu, `debug-layer.h/cpp` has been split into multiple smaller files organized around individual classes.
- Added a new `debug-layer` folder containing the post-split files and updated premake to account for it

### CODE REVIEW

File: tools/gfx/cpu/cpu-helper-functions.h [hunk 1]
```diff
@@ -6,6 +6,6 @@ namespace gfx
 {
 using namespace Slang;
 
-Result SLANG_MCALL createCPUDevice(const IDevice::Desc* desc, IDevice** outDevice);
+
```
Comment (csyonghe) [inline tools/gfx/cpu/cpu-helper-functions.h:9]: Can remove this file.

File: tools/gfx/cpu/cpu-helper-functions.cpp [hunk 2]
```diff
@@ -7,12 +7,6 @@ namespace gfx
...
 using namespace Slang;
 
-Result SLANG_MCALL createCPUDevice(const IDevice::Desc* desc, IDevice** outDevice)
-{
-    RefPtr<cpu::DeviceImpl> result = new cpu::DeviceImpl();
-    SLANG_RETURN_ON_FAIL(result->initialize(*desc));
-    returnComPtr(outDevice, result);
-    return SLANG_OK;
-}
+
```
Comment (csyonghe) [inline tools/gfx/cpu/cpu-helper-functions.cpp:10]: We don't need this file.

**Review by csyonghe — APPROVED** [review]

File: tools/gfx/cpu/cpu-helper-functions.h (hunk 1 above)
Comment (lucy96chen) [reply tools/gfx/cpu/cpu-helper-functions.h:9]: Oops, will remove these and submit whenever the shader cache work is ready for a PR.

File: tools/gfx/cpu/cpu-helper-functions.cpp (hunk 2 above)
Comment (lucy96chen) [reply tools/gfx/cpu/cpu-helper-functions.cpp:10]: Same as above

**Pull request merged**